general/arrays_by_name.hpp

```diff
@@ -0,0 +1,256 @@
+   /// possible typos or other errors. To handle such errors more gracefully
+   /// consider first calling EntryExists.
+   inline Array<T> &operator[](const std::string &name);
+
+   /// @brief Const reference access to the named entry.
+   ///
+   /// @note Passing a name for a nonexistent array will print an error
+   /// message and halt execution. This is intended to call attention to
+   /// possible typos or other errors. To handle such errors more gracefully
+   /// consider first calling EntryExists.
+   inline const Array<T> &operator[](const std::string &name) const;
+
+   /// @brief Create a new empty array with the given name
+   ///
+   /// @note Passing a name for an already existent array will print an error
+   /// message and halt execution. This is intended to call attention to
+   /// possible typos or other errors. To handle such errors more gracefully
+   /// consider first calling EntryExists.
+   inline Array<T> &CreateArray(const std::string &name);
+
+   /// @brief Delete all named arrays from the container
+   inline void DeleteAll();
+
+   /// @brief Delete the named array from the container
+   ///
+   /// @note Passing a name for a nonexistent array will print an error
+   /// message and halt execution. This is intended to call attention to
+   /// possible typos or other errors. To handle such errors more gracefully
+   /// consider first calling EntryExists.
+   inline void DeleteArray(const std::string &name);
+
+   /// @brief Copy assignment operator: deep copy from 'src'.
+   ArraysByName<T> &operator=(const ArraysByName<T> &src) = default;
+
+   /// @brief Move assignment operator: deep move from 'src'.
+   ArraysByName<T> &operator=(ArraysByName<T> &&src) noexcept = default;
+
+   /// @brief Assignment operator (deep copy) from @a src, an ArraysByName
+   /// container of convertible type.
+   template <typename CT>
+   inline ArraysByName &operator=(const ArraysByName<CT> &src);
+
+   /// @brief Print the contents of the container to an output stream
+   ///
+   /// @note By default each array will be printed on its own line. A specific
+   /// number of entries per line can be used by changing the @a width argument.
+   void Print(std::ostream &out = mfem::out, int width = -1) const;
+
+   /// @brief Load the contents of the container from an input stream
+   ///
+   /// @note This method will not first empty the container. First call
+   /// DeleteAll if this behavior is needed.
+   void Load(std::istream &in);
+
+   /// @brief Sort each named array in the container
+   inline void SortAll();
+
+   /// @brief Remove duplicates from each sorted named array
+   ///
+   /// @note Identical entries may exist in multiple arrays but will only occur
+   /// at most once in each array.
+   inline void UniqueAll();
+
+   /// STL-like begin.  Returns pointer to the first element of the array.
+   inline iterator begin() { return data.begin(); }
```

Comment:
```suggestion
   iterator begin() { return data.begin(); }
```
And similarly for the `begin` and `end` below.